### PR TITLE
Add metadata generation for Java core proto task.

### PR DESCRIPTION
**Pull request by shinfan**
This PR adds support to generate package metadata files for Java through Artman (Packman free)

Toolkit PR: https://github.com/googleapis/toolkit/pull/1133
googleapis PR: googleapis/googleapis#296

### CODE REVIEW

File: resources/java/LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
+                            Apache License
```
Comment (geigerj) [inline resources/java/LICENSE]: I think all resources in this directory be in `toolkit`, not `artman`: I'd consider this part of the gRPC package metadata generation.

File: pipeline/tasks/protoc_tasks.py [hunk 2]
```diff
@@ -262,6 +266,15 @@ def _prepare_pkg_dir(output_dir, api_name, api_version, organization_name,
         stderr=subprocess.STDOUT)
     return pkg_dir
 
+def _check_protoc_version(language):
```
Comment (geigerj) [inline pipeline/tasks/protoc_tasks.py]: I don't think validating what version of protoc should be the business of an Artman task; we build the Docker images for situations where you want to enforce consistent installation of dependencies. (And it's brittle to validate by checking against a hardcoded string that we expect the protoc binary to print to stdout.)

File: resources/java/LICENSE (hunk 1 above)
Comment (garrettjonesgoogle) [inline resources/java/LICENSE]: +2

Comment (shinfan) [inline resources/java/LICENSE]: Done

File: pipeline/tasks/protoc_tasks.py (hunk 2 above)
Comment (shinfan) [inline pipeline/tasks/protoc_tasks.py]: Fair enough. Removed.

File: pipeline/tasks/protoc_tasks.py [hunk 3]
```diff
@@ -192,6 +192,10 @@ def proto_compiler_command(self):
     'python': _PythonProtoParams()
 }
 
+_PROTOC_VERSION_MAP = {
```
Comment (geigerj) [inline pipeline/tasks/protoc_tasks.py]: I think this file can be reverted

Comment (shinfan) [inline pipeline/tasks/protoc_tasks.py]: Sorry I was not fully pushed my local changes. Now this is fixed.

**Comment by shinfan** [comment]
@geigerj  PTAL

File: pipeline/tasks/protoc_tasks.py [hunk 4]
```diff
@@ -540,8 +541,17 @@ def execute(self, src_proto_path, api_name,
...
+        self.exec_command([
+            'cp', '-rf', os.path.join(intermediate_package_dir, 'src'), package_dir])
+
+        # Clean up non-proto files
         for root, dirs, files in os.walk(proto_output_dir):
             for file_ in files:
                 if not os.path.splitext(file_)[1] == '.proto':
                     os.remove(os.path.join(root, file_))
+
+        # Remove intermidiate output
```
Comment (garrettjonesgoogle) [inline pipeline/tasks/protoc_tasks.py]: spelling (intermidiate)

File: pipeline/tasks/protoc_tasks.py [hunk 5]
```diff
@@ -540,8 +541,17 @@ def execute(self, src_proto_path, api_name,
             self.exec_command(['mkdir', '-p', output_path])
             self.exec_command([
                 'cp', '-rf', proto_path, output_path])
-        # Clean up non proto files
+
+        # Copy compiled proto files
```
Comment (garrettjonesgoogle) [inline pipeline/tasks/protoc_tasks.py]: For java, does this mean ".java" files or ".class" files? The "compiled" here feels ambiguous. Is it correct to say "Copy files generated from protos"?

Comment (shinfan) [inline pipeline/tasks/protoc_tasks.py]: Those are .java files. "Copy files generated from protos" sounds right to me.

Comment (shinfan) [inline pipeline/tasks/protoc_tasks.py]: Done

File: pipeline/tasks/protoc_tasks.py (hunk 4 above)
Comment (shinfan) [inline pipeline/tasks/protoc_tasks.py]: Done

**Comment by shinfan** [comment]
@garrettjonesgoogle  PTAL

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**